### PR TITLE
new package: py-batchspawner

### PR DESCRIPTION
**Pull request by healther**
Part of our recent merge and upstream effort

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-batchspawner/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+from spack import *
+
+
+class PyBatchspawner(PythonPackage):
+    """This is a custom spawner for Jupyterhub that is designed for
+    installations on clusters using batch scheduling software."""
+
+    homepage = "https://github.com/jupyterhub/batchspawner"
+    url      = "https://pypi.io/packages/source/b/batchspawner/batchspawner-1.0.1.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-batchspawner/package.py]:
```suggestion
    pypi     = "batchspawner/batchspawner-1.0.1.tar.gz"
```

File: var/spack/repos/builtin/packages/py-batchspawner/package.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    version('1.0.1', sha256='b96ab7e1eb3b69e0863ebf045b960a4d074935a8a8fbfd0369b5d1af5b1bab8d')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-batchspawner/package.py:17]:
```python
depends_on('python@3.3:3.999', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
PyPI tarball seems to be missing the `requirements.txt` file, does this package actually work?

File: var/spack/repos/builtin/packages/py-batchspawner/package.py (hunk 1 above)
Comment (healther) [reply var/spack/repos/builtin/packages/py-batchspawner/package.py]: Done

File: var/spack/repos/builtin/packages/py-batchspawner/package.py (hunk 2 above)
Comment (healther) [reply var/spack/repos/builtin/packages/py-batchspawner/package.py:17]: Done

**Comment by healther** [comment]
> PyPI tarball seems to be missing the `requirements.txt` file, does this package actually work?

Test building to be sure, but yes, we used it at some point (albeit not in the live-container, so I cannot just dump it)

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
License needs to be updated (I think it's just a year change) to pass the tests.

**Comment by healther** [comment]
> License needs to be updated (I think it's just a year change) to pass the tests.

will do after the test build ran through, I just got a checksum fail of `py-jupyter-client` that I need to investigate. It should :tm: just :tm: be an outdated cache of us :tm: but...

**Comment by healther** [comment]
You were right, a local bugfix did not make it to our master branch, now builds as is

File: var/spack/repos/builtin/packages/py-batchspawner/package.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+from spack import *
+
+
+class PyBatchspawner(PythonPackage):
+    """This is a custom spawner for Jupyterhub that is designed for
+    installations on clusters using batch scheduling software."""
+
+    homepage = "https://github.com/jupyterhub/batchspawner"
+    pypi     = "batchspawner/batchspawner-1.0.1.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-batchspawner/package.py]:
```suggestion
    pypi     = "batchspawner/batchspawner-1.1.0.tar.gz"
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-batchspawner/package.py (hunk 3 above)
Comment (healther) [reply var/spack/repos/builtin/packages/py-batchspawner/package.py]: Done, but I thought these kind of offsets don't matter. Has this changed?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**